.github/actions/common/predicate-utils/src/index.ts

```diff
@@ -0,0 +1,161 @@
+  externalParameters: { [key: string]: ParameterValue };
+
+  // systemParameters describes parameters of the build environment provided by the `builder`.
+  systemParameters?: { [key: string]: ParameterValue };
+
+  // resolvedDependencies are dependencies needed at build time.
+  resolvedDependencies?: ArtifactReference[];
+}
+
+export interface RunDetails {
+  builder: Builder;
+
+  metadata: Metadata;
+
+  byproducts?: ArtifactReference[];
+}
+
+export interface SLSAv1Predicate {
+  // buildDefinition describes the inputs to the build.
+  buildDefinition: BuildDefinition;
+
+  // runDetails includes details specific to this particular execution of the build.
+  runDetails: RunDetails;
+}
+
+// addGitHubContext
+export function addGitHubContext(
+  predicate: SLSAv1Predicate,
+  currentRun: ApiWorkflowRun
+): SLSAv1Predicate {
+  const { env } = process;
+
+  if (!predicate.buildDefinition.systemParameters) {
+    predicate.buildDefinition.systemParameters = {};
+  }
+
+  // TODO(https://github.com/slsa-framework/slsa-github-generator/issues/1505):
+  // Add GitHub event payload.
+  const systemParams = predicate.buildDefinition.systemParameters;
+  systemParams["GITHUB_EVENT_NAME"] = { value: env.GITHUB_EVENT_NAME || "" };
+  systemParams["GITHUB_JOB"] = { value: env.GITHUB_JOB || "" };
+  systemParams["GITHUB_REF"] = { value: env.GITHUB_REF || "" };
+  systemParams["GITHUB_REF_TYPE"] = { value: env.GITHUB_REF_TYPE || "" };
+  systemParams["GITHUB_REPOSITORY"] = { value: env.GITHUB_REPOSITORY || "" };
+  systemParams["GITHUB_RUN_ATTEMPT"] = { value: env.GITHUB_RUN_ATTEMPT || "" };
+  systemParams["GITHUB_RUN_ID"] = { value: env.GITHUB_RUN_ID || "" };
+  systemParams["GITHUB_RUN_NUMBER"] = { value: env.GITHUB_RUN_NUMBER || "" };
+  systemParams["GITHUB_SHA"] = { value: env.GITHUB_SHA || "" };
+  systemParams["GITHUB_WORKFLOW"] = { value: env.GITHUB_WORKFLOW || "" };
+  systemParams["GITHUB_ACTOR_ID"] = {
+    value: String(currentRun.actor?.id || ""),
+  };
+  systemParams["GITHUB_REPOSITORY_ID"] = {
+    value: String(currentRun.repository.id || ""),
+  };
+  systemParams["GITHUB_REPSITORY_OWNER_ID"] = {
+    value: String(currentRun.repository.owner.id || ""),
+  };
+  systemParams["GITHUB_WORKFLOW_REF"] = {
+    value: env.GITHUB_WORKFLOW_REF || "",
+  };
+  systemParams["GITHUB_WORKFLOW_SHA"] = {
+    value: env.GITHUB_WORKFLOW_SHA || "",
+  };
+  systemParams["IMAGE_OS"] = { value: env.ImageOS || "" };
+  systemParams["IMAGE_VERSION"] = { value: env.ImageVersion || "" };
+  systemParams["RUNNER_ARCH"] = { value: env.RUNNER_ARCH || "" };
+  systemParams["RUNNER_NAME"] = { value: env.RUNNER_NAME || "" };
+  systemParams["RUNNER_OS"] = { value: env.RUNNER_OS || "" };
+
+  if (env.GITHUB_EVENT_NAME === "workflow_dispatch") {
+    if (env.GITHUB_EVENT_PATH) {
+      const ghEvent: WorkflowDispatchEvent = JSON.parse(
+        fs.readFileSync(env.GITHUB_EVENT_PATH).toString()
+      );
+
+      for (const input in ghEvent.inputs) {
```

Comment:
this works for generators, but is that correct for BYOB / builders?
IIUC, for builders, we need to record the TRW's inputs, which are passed to the `setup-token` Action via `slsa-workflow-inputs` field. I think we need to record those.

We temporarily need to read the top-workflow inputs (like in the code) and identify the entries that are not TRW entries, in order to fix https://github.com/actions/runner/issues/2274

Let me know if I mis-understood something

Reply:
Here's something I've been thinking about regarding this: 
* Let's keep workflow inputs (the ones extracted here) into systemParameters for ALL workflows. This way this particular function can work solely on modifying systemParmaters to include GH context vars for any time of GH workflow.
* For delegators, we'll add in it's own specific logic to add TRW inputs, since that's a "buildType = delegator" specific thing.

Reply:
> Let's keep workflow inputs (the ones extracted here) into systemParameters for ALL workflows

Should this be under `externalParameters` for non-delegator workflows as well? The generic generator *can* report the workflow inputs under `externalParameters` as well to follow v1.0?

Reply:
The problem is what is considered "user inputs" for either build type. the workflow inputs or the TRW inputs? If the delegator is attesting to a TRW, then I think the TRW inputs are actually the "top level" and the workflow inputs are the system's parameters.

I think the resolution is this common library can expose `GetInputs` instead, and leave this for the user to decide.

Reply:
that makes sense. So the generic generator could use `GetInputs` to set their own `externalParameters`, whereas the delegator would not use the `GetInputs`.

Reply:
This is the TODO as well: https://github.com/slsa-framework/slsa-github-generator/issues/1504

I'll table this for an in person discussion, would be good to get the story clear for both types of generators.